ignite/contrib/handlers/tqdm_logger.py

```diff
@@ -270,23 +266,17 @@ def __call__(self, engine: Engine, logger: ProgressBar, event_name: Union[str, E
             desc += f" [{global_step}/{max_num_of_closing_events}]"
         logger.pbar.set_description(desc)  # type: ignore[attr-defined]
 
-        metrics = self._setup_output_metrics(engine)
-
-        rendered_metrics = OrderedDict()  # type: Dict[str, Union[str, float, numbers.Number]]
-        for key, value in metrics.items():
-            if isinstance(value, numbers.Number) or isinstance(value, str):
-                rendered_metrics[key] = value
-            elif isinstance(value, torch.Tensor) and value.ndimension() == 0:
-                rendered_metrics[key] = value.item()
-            elif isinstance(value, torch.Tensor) and value.ndimension() == 1:
-                for i, v in enumerate(value):
-                    k = f"{key}_{i}"
-                    rendered_metrics[k] = v.item()
-            else:
-                warnings.warn(f"ProgressBar can not log tensor with {value.ndimension()} dimensions")
-
-        if rendered_metrics:
-            logger.pbar.set_postfix(rendered_metrics)  # type: ignore[attr-defined]
+        rendered_metrics = self._setup_output_metrics(engine, log_text=True)
+        metrics = OrderedDict()
+        for key, value in rendered_metrics.items():
+            keys_list = key.split("/")
+            keys_list.pop(0)  # tqdm has tag as description
+            key = "_".join(keys_list)
```

Comment:
I was wondering if it would be better instead of doing split by "/" to have keys as tuple of `(tag, name)` or `(tag, name, i)` such that we can create appropriate string for each logger by `"/".join(key)` ?

Reply:
Yeah, can do that although since most of them follow the "/" separated convention (like Polyaxon, visdom and wandb) the dict can be directly passed to the logger without the need of any loops in the derived class. If the uniform tuple method is more useful I can go with that.

Reply:
Yes, I also thought about that, but splitting the key looks like patching. Let's try to use tuples.

Reply:
Yeah, fixed!